src/Typography.js

```diff
@@ -0,0 +1,19 @@
+/** @jsx jsx */
+import { jsx } from '@emotion/core'
+import _ from 'lodash/fp'
+import { fonts } from './config'
+import { findKeys } from './utils'
+
+let getVariants = (props, variants) =>
+  _.flow(
+    _.pick(findKeys(_.stubTrue, props)),
+    _.values
+  )(variants)
+
+let toComponent = ({ as: As = 'div', variants, ...styles }) => props => (
+  <As css={[styles, ...getVariants(props, variants)]} {...props} />
```

Comment:
It might be worth documenting the precedence order - I’m pretty sure whatever variants are declared last will be the ones that end up being spread last and therefore taking precedence if multiple variants are passed in